### PR TITLE
@ #11 | should add docs how to get SSO work with Dex

### PR DESCRIPTION
**Pull request by hoatle**
connect #11 

assigned to @hieptranquoc to review the technical content, @hoavt to review the English grammar.

### CODE REVIEW

**Comment by hieptranquoc** [comment]
I've reviewed! @hoavt please check grammar and review

File: docs/sso.md [hunk 1]
```diff
@@ -0,0 +1,392 @@
+# SSO (Single Sign On with Kubernetes)
+
+This guide will help you to set up SSO for k8s. You should set up a local k8s to test this on local
+by following https://github.com/teracyhq-incubator/teracy-dev-entry-k8s#how-to-use
```
Comment (hoavt) [inline docs/sso.md]: should have a full stop at the end of the sentence

File: docs/sso.md [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+
+We're going to use Dex as a OIDC provider for k8s authentication, you can use any other OIDC providers
+with the similar deployment steps.
+
+
+## Domain Aliases
+
+We're going to deploy dex (auth.k8s.local) and dex-k8s-authenticator (login.k8s.local) helm charts,
+so make sure to configure these domain aliases (see https://github.com/teracyhq-incubator/teracy-dev-entry-k8s#domain-aliases)
```
Comment (hoavt) [inline docs/sso.md]: should have a full stop at the end of the sentence

File: docs/sso.md [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+
+Then deploy it:
+
+
+```
+$ kubectl create namespace dex # if dex namespace not created yet
+$ helm upgrade --install --namespace dex dex stable/dex -f dex.yaml
+```
+
+After that, `https://auth.k8s.local/.well-known/openid-configuration` should display info such as:
```
Comment (hoavt) [inline docs/sso.md]: display the info like this:

File: docs/sso.md [hunk 4]
```diff
@@ -0,0 +1,392 @@
...
+    "locale",
+    "name",
+    "sub"
+  ]
+}
+```
+
+`$ kubectl -n dex get configmaps dex-web-server-ca -o yaml` to see the dex CA root certifcate that
+the k8s api server must trust and the dex client apps must trust when the self-signed certificate is
+used for the dex deployment. Create `sso/dex-ca.pem` file from the config map data above.
```
Comment (hoavt) [inline docs/sso.md]: Create the `sso/dex-ca.pem` file

File: docs/sso.md [hunk 5]
```diff
@@ -0,0 +1,392 @@
...
+
+
+## Configure the k8s api server
+
+Set the ansible config, through the `workspace/inventory/group_vars/k8s-cluster/k8s-cluster.yaml` or though the teracy-dev-k8s host_vars config:
```
Comment (hoavt) [inline docs/sso.md]: config --> configuration

File: docs/sso.md [hunk 6]
```diff
@@ -0,0 +1,392 @@
...
+
+The `dex-ca.crt` value is base64 encoded from the `dex-ca.pem` file.
+
+After that:
+
+```
+helm upgrade --install --namespace dex login charts/dex-k8s-authenticator -f dex-k8s-authenticator.yaml
+```
+
+and open https://login.k8s.local for the k8s login instruction
```
Comment (hoavt) [inline docs/sso.md]: Should have a full stop at the end of the sentence.

**Pull request merged**